include/ccf/pal/snp_ioctl6.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/pal/attestation_sev_snp.h"
+
+#include <fcntl.h>
+#include <stdint.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+// Based on the SEV-SNP ABI Spec document at
+// https://www.amd.com/system/files/TechDocs/56860.pdf
+
+/* linux kernel 6.* versions of the ioctls that talk to the PSP */
+
+namespace ccf::pal::snp::ioctl6
+{
+  constexpr auto DEVICE = "/dev/sev-guest";
+
+  // Table 20
+  struct AttestationReq
+  {
+    uint8_t report_data[snp_attestation_report_data_size];
+    uint32_t vmpl;
+    uint8_t reserved[28]; // needs to be zero
+  }; // aka snp_report_req in (linux) include/uapi/linux/sev-guest.h
+
+  // Table 23
+#pragma pack(push, 1)
+  struct AttestationResp
+  {
+    uint32_t status;
+    uint32_t report_size;
+    uint8_t reserved[0x20 - 0x8];
+    struct Attestation report;
+    uint8_t padding[64];
+    // padding to the size of SEV_SNP_REPORT_RSP_BUF_SZ (i.e., 1280 bytes)
+  };
```

Comment:
Yes, those are unchanged for our purposes. I don't think this makes a huge difference, because we will drop the 5.x one once C-ACI is done upgrading but sure.